server/src/test/java/io/spine/server/event/EventStoreShould.java

```diff
@@ -210,21 +221,102 @@ public void fail_to_store_events_of_different_tenants_in_a_single_operation() {
+                                                            .build();
         final EventContext secondTenantContext = EventContext.newBuilder()
-                                                           .setCommandContext(secondTenantCommand)
-                                                           .build();
+                                                             .setCommandContext(secondTenantCommand)
+                                                             .build();
         final Event firstTenantEvent = Event.newBuilder()
                                             .setContext(firstTenantContext)
                                             .build();
         final Event secondTenantEvent = Event.newBuilder()
-                                            .setContext(secondTenantContext)
-                                            .build();
+                                             .setContext(secondTenantContext)
+                                             .build();
         final Collection<Event> event = ImmutableSet.of(firstTenantEvent, secondTenantEvent);
         eventStore.appendAll(event);
     }
 
+    @Test
+    public void not_store_enrichment_for_EventContext() {
+        final Event event = projectCreated(Time.getCurrentTime());
+        final Event enriched = event.toBuilder()
+                                    .setContext(event.getContext()
+                                                     .toBuilder()
+                                                     .setEnrichment(newEnrichment()))
+                                    .build();
+        eventStore.append(enriched);
+        final MemoizingObserver<Event> observer = memoizingObserver();
+        eventStore.read(EventStreamQuery.getDefaultInstance(), observer);
+        final EventContext context = observer.responses()
+                                             .get(0)
+                                             .getContext();
+        assertTrue(isDefault(context.getEnrichment()));
+    }
+
+    @Test
+    public void not_store_enrichment_for_origin_of_RejectionContext_type() {
+        final RejectionContext origin = RejectionContext.newBuilder()
+                                                        .setEnrichment(newEnrichment())
+                                                        .build();
+        final Event event = projectCreated(Time.getCurrentTime());
+        final Event enriched = event.toBuilder()
+                                    .setContext(event.getContext()
+                                                     .toBuilder()
+                                                     .setRejectionContext(origin))
+                                    .build();
+        eventStore.append(enriched);
+        final MemoizingObserver<Event> observer = memoizingObserver();
+        eventStore.read(EventStreamQuery.getDefaultInstance(), observer);
+        final RejectionContext modifiedOrigin = observer.responses()
```

Comment:
What did you mean by `modifiedOrigin`? This question applies to this variable and similar below.